### PR TITLE
Refactor FocusComponent and Health (non-OSGi).'

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/Main.java [hunk 1]
```diff
@@ -38,54 +41,133 @@
...
+        {
+            WaitableBundleActivator.waitUntilStarted();
+        }
+        catch (Exception e)
+        {
+            logger.error("Failed to start all OSGi bundles, exiting.");
+            OSGi.stop(activator);
+            return;
+        }
+        logger.warn("OSGi services started.");
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/Main.java:88]: Log on info?

File: src/main/java/org/jitsi/jicofo/Main.java [hunk 2]
```diff
@@ -38,54 +41,133 @@
...
+            public void stop(BundleContext bundleContext)
+                    throws Exception
+            {
+                // We're doing nothing
+            }
+        };
+
+
+        // Start OSGi
+        logger.warn("Starting Osgi");
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/Main.java:160]: info?

File: src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java [hunk 3]
```diff
@@ -80,67 +83,60 @@
...
-    /**
-     * Releases resources used by this instance.
-     */
-    public void dispose()
+    public void connect()
     {
-        OSGi.stop(this);
+        if (!isPingTaskStarted())
+        {
+            startPingTask();
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java:124]: Does this need to be atomic with the `isPingTaskStarted()` check?

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java:124]: In theory -- yes. In practice -- no. That's the only place any of that code is used, it [has always been](https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java#L118) like this, and I'm hoping to remove all that code soon, so I'd rather not add to it.

Comment (bbaldino) [reply src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java:124]: Cool, just checking.

**Review by bbaldino — APPROVED** [review]

**Pull request merged**